docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,106 @@
+          * Create a commit that updates our version number to something like `[firefox-version].0-nightly-[YYYYMMDD]`, similar to what `prepare-release.py` does now.
+            * Make sure that the github CI can push this commit without needing PR approval.
+            * Note: we will no longer be using semver-style versions and also not following semver semantics.  We will be releasing breaking changes in a nightly cycle without bumping the minor version.
+          * Create a GitHub release from the branch with a tag based on the version number / date (`v[firefox-version].0-nightly-[YYYYMMDD]`). This will cause maven packages to be published, start the `rust-components-swift` CI, etc.
+        * Manually handle releases from the `release_v*` branches whenever new code is committed to those branches
+          * Update our version number to `[firefox-version].[N+1]`
+          * Create a GitHub release from the branch
+          * Create PRs to merge the changes forward to `main` and more recent release branches.
+        * Continue to maintain our CHANGELOG, but with new versioning.  In main, keep a rolling log of updates with the header `[firefox-version].0`
+        * This replaces our current release process -- no more cutting releases.
+        * To push changes to a consumer: create a PR in their repository that bumps the application-services version to an existing release.
+        * On the monthly Firefox merge date:
+          * Create a `release_v[N]` branch of of main
+          * Create a GitHub release for that branch
+          * Bump the major version number in main and start a new changelog header.
+    * Phase 2: Automate nightly merges of `main` into consumer applications (March 2023 - May 2023)
+      * Whenever a new application-services version is released (including nightly, beta, release, and ESR versions).
+        * For each consumer application (including all mobile apps and Desktop):
+            * Create a PR to take the new application-services version.
+            * Set up an auto-merge system where if the test suite passes we automatically merge the PR.
+            * Set up an alert that gets sent to `sync-bots` if the test suite fails.  This means we would need to identify why the tests failed and assign a developer to resolve the issue.  The expectation would be that the developer fixes things in time for the next nightly merge.
+    * All application-services changes that are breaking for consumers should have corresponding PRs in the consumer repos that have been reviewed and approved.  If we accidentally merge the change into application-services before the PRs are ready, then we should back the change out.
+
+## Decision Outcome
+
+**(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+
+## Pros and Cons of the Options
+
+### Keep our current release process
+
+  * Good, because it doesn't require changes to our workflow
```

Comment:
We should enumerate what's bad about this - the way this section reads it almost sounds like we should stick with the status quo (status quo is all "good", changed workflow has many "bad" :)

Probably just something like "* Bad, for the reasons enumerated at the top of this document" or similar?